### PR TITLE
Added tons of documentation and cleaned up some code

### PR DESCRIPTION
**Pull request by Swofty-Developments**

### CODE REVIEW

**Comment by Swofty-Developments** [comment]
PR changes Java version to 21 in order to remove the need for preview flags, this will mean that the Git actions will have to be updated accordingly.

File: src/main/java/me/replydev/qubo/CLI.java [hunk 1]
```diff
@@ -5,68 +5,87 @@
...
     }
 
+    /**
+     * Checks if UTF-8 mode is enabled for file encoding.
+     * @return true if UTF-8 mode is set, false otherwise.
+     */
     private boolean isUTF8Mode() {
         List<String> arguments = ManagementFactory.getRuntimeMXBean().getInputArguments();
-        return arguments.contains("-Dfile.encoding=UTF-8");
+        return arguments.stream().anyMatch(arg -> arg.equals("-Dfile.encoding=UTF-8"));
```
Comment (replydev) [inline src/main/java/me/replydev/qubo/CLI.java:89]: Why? It should be slower.

**Review by replydev — CHANGES_REQUESTED** [review]
Hi and thanks a lot for your contribution. Just one thing and we are good to go.

**Review by replydev — APPROVED** [review]

**Comment by Swofty-Developments** [comment]
Thanks

**Pull request merged**